packetbeat/protos/dns/dns.go

```diff
@@ -551,6 +551,30 @@ func rrsToMapStrs(records []mkdns.RR) []common.MapStr {
 	return mapStrArray
 }
 
+// ansToMapStrs extracts answers from an array of RR's and returns MapStr.
+func ansToMapStrs(answers []mkdns.RR) common.MapStr {
+        answer_class := []string{}
+        answer_data := []string{}
+        answer_name := []string{}
+        answer_ttl := []string{}
+        answer_type := []string{}
+
+        for _, a := range rrsToMapStrs(answers) {
+            answer_class = append(answer_class, fmt.Sprint(a["class"]))
+            answer_data = append(answer_data, fmt.Sprint(a["data"]))
+            answer_name = append(answer_name, fmt.Sprint(a["name"]))
+            answer_ttl = append(answer_ttl, fmt.Sprint(a["ttl"]))
+            answer_type = append(answer_type, fmt.Sprint(a["type"]))
```

Comment:
There are a bunch of other fields fields that could possibly be present in each answer. They would need to be included.

It seems like this could lead to some sparsely populated arrays (assuming we want to maintain the relationship between fields in a RR using the index). For example, if the a response contains ten NS records and one RRSIG record, what does the `dns.answers.algorithm` field look like? Is it an an array of ten nulls followed by the algorithm value from the lone RRSIG record?